application/core/src/main/java/com/github/pieter_groenendijk/controller/ReservationController.java

```diff
@@ -1,47 +1,58 @@
+import com.github.pieter_groenendijk.repository.IReservationRepository;
+import com.github.pieter_groenendijk.repository.ReservationRepository;
 import com.github.pieter_groenendijk.service.IReservationService;
 import com.github.pieter_groenendijk.service.ReservationService;
 import io.swagger.v3.oas.annotations.Operation;
 import io.swagger.v3.oas.annotations.responses.ApiResponse;
 import io.swagger.v3.oas.annotations.responses.ApiResponses;
+import org.hibernate.SessionFactory;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
 
-import java.util.Date;
+
 
 @RestController
-@RequestMapping("/api/reservations")
+@RequestMapping("/reservation")
 public class ReservationController {
 
-    private IReservationService reservationService;
-
+    private final SessionFactory sessionFactory = new SessionFactoryFactory().create();
+    private final IReservationService reservationService;
+    IMembershipRepository membershipRepository;
 
+    public ReservationController() {
+        IReservationRepository reservationRepository = new ReservationRepository(sessionFactory);
+        reservationService = new ReservationService(reservationRepository, membershipRepository);
+    }
     @Operation(summary = "Create a reservation", description = "Create a new reservation")
     @ApiResponses(value = {
             @ApiResponse(responseCode = "201", description = "Reservation created")
     })
     @PostMapping
-    public ResponseEntity<Reservation> store(@RequestParam long membershipId, @RequestParam long copyId) {
-        Reservation reservation = reservationService.store(membershipId, copyId);
+    public ResponseEntity<Reservation> store(@RequestParam("MembershipId") long membershipId, @RequestParam("ProductCopyId")  long productCopyId) {
+        Reservation reservation = reservationService.store(membershipId, productCopyId);
         return new ResponseEntity<>(reservation, HttpStatus.CREATED);
     }
 
+
     @Operation(summary = "Get all reservation details by reservationId", description = "Get reservation details by reservationId")
     @ApiResponses(value = {
             @ApiResponse(responseCode = "200", description = "Reservation found"),
             @ApiResponse(responseCode = "204", description = "No reservation found for the given reservationId\"")
     })
     @GetMapping("/{reservationId}")
-    public ResponseEntity<Reservation> getReservationById(@PathVariable long reservationId) {
+    public ResponseEntity<Reservation> getReservationById(@PathVariable ("reservationId")long reservationId) {
         Reservation reservation = reservationService.getReservationById(reservationId);
         if (reservation != null) {
             return new ResponseEntity<>(reservation, HttpStatus.OK);
         } else {
-            return new ResponseEntity<>(null, HttpStatus.OK);
+            return new ResponseEntity<>(null, HttpStatus.NO_CONTENT);
```

Comment:
De afhandeling van unhappy paths zijn anders dan in mijn stromen. Denk niet dat 1 van beide fout is. Veel toestand om dit nu allemaal te gaan herschrijven, dus we laten het zo. Mochten we het later nog willen refactoren, dan kan dat altijd nog.